tls/extensions/s2n_extension_type_lists.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include "tls/extensions/s2n_extension_type.h"
+
+typedef enum {
+    S2N_EXTENSION_LIST_CLIENT_HELLO = 0,
+    S2N_EXTENSION_LIST_SERVER_HELLO,
+    S2N_EXTENSION_LIST_ENCRYPTED_EXTENSIONS,
+    S2N_EXTENSION_LIST_CERT_REQ,
+    S2N_EXTENSION_LIST_CERTIFICATE,
+} s2n_extension_list_id;
```

Comment:
Would it be useful to add a final element
`S2N_EXPENSIONS_LIST_NUM_ELEMENTS = S2N_EXTENSION_LIST_CERTIFICATE`

Reply:
I haven't found a use for a `_NUM_ELEMENTS` enum yet, so I didn't add one.